http/jsonapi/runtime/standart_params.go

```diff
@@ -63,46 +60,86 @@ func (m *MapMapper) Map(value string) (string, bool) {
 	return val, isValid
 }
 
-// PaginationFromRequest extracts pagination query parameter and  returns a function that adds the pagination to a query
-func PaginationFromRequest(r *http.Request) (QueryOption, error) {
+// UrlQueryParameters contains all information that are needed for pagination, sorting and filtering.
+// It is not depending on orm.Query
+type UrlQueryParameters struct {
+	HasPagination bool
+	PageNr        int
+	PageSize      int
+	Order         []string
+	Filter        map[string][]interface{}
+}
+
+// ReadURLQueryParameters reads sorting, filter and pagination from requests and return a UrlQueryParameters object,
+// even if any errors occur. The returned error combines all errors of pagination, filter and sorting.
+func ReadURLQueryParameters(r *http.Request, mapper ColumnMapper, sanitizer ValueSanitizer) (*UrlQueryParameters, error) {
+	result := &UrlQueryParameters{}
+	errPagination := result.setPagination(r)
+	errSorting := result.setSorting(r, mapper)
+	errFilter := result.setFilter(r, mapper, sanitizer)
+	if errPagination != nil || errSorting != nil || errFilter != nil {
+		err := fmt.Errorf("problems occured while ready filter, sorting or pagination from request: filter: %w, sorting: %w, pagination: %w", errFilter, errSorting, errPagination)
+		return result, err
+	}
+	return result, nil
+}
+
+// AddToQuery adds filter, sorting and pagination to a orm.Query
+func (u *UrlQueryParameters) AddToQuery(query *orm.Query) *orm.Query {
+	if u.HasPagination {
+		if u.PageNr == 0 {
+			query.Offset(0)
+		} else {
+			query.Offset((u.PageSize * u.PageNr) - 1)
+		}
+		query.Limit(u.PageSize)
+	}
+	for name, filterValues := range u.Filter {
+		if len(filterValues) == 0 {
+			continue
+		}
+
+		if len(filterValues) == 1 {
+			query.Where(name+" = ?", filterValues[0])
+			continue
+		}
+		query.Where(name+" IN (?)", pg.In(filterValues))
+	}
+	for _, val := range u.Order {
+		query.Order(val)
+	}
+	return query
+
+}
+
+func (u *UrlQueryParameters) setPagination(r *http.Request) error {
 	pageStr := r.URL.Query().Get("page[number]")
 	sizeStr := r.URL.Query().Get("page[size]")
 	if pageStr == "" || sizeStr == "" {
```

Comment:
Can't we have a default page size? Does the jsonapi spec require both parameter to be set?

I would suggest that we allow pagination with "only" the page number.

Reply:
e.g., we could use `cfg.MinPageSize` if none is set?